build/Build.cs

```diff
@@ -696,6 +682,27 @@ string GetNugetVersion()
                   ?? throw new InvalidOperationException("Unable to retrieve valid Nuget Version");
         }
 
+        void CreateHashFileForProject(string directory)
+        {
+            byte[] GetHashOfFileContent(string f)
+            {
+                using var md5 = MD5.Create();
+                using var stream = File.OpenRead(f); 
+                return md5.ComputeHash(stream);
+            }
+
+            var allFiles = Directory.GetFiles(directory, "*.*", SearchOption.AllDirectories);
```

Comment:
QQ is this recursive?

Reply:
yeah it is - I figured we should to drill into all sub directories to collect the required hashes - might be slow? or is there another concern?
(btw - this results in the same file count as seen from the terminal, so logically - it does what I thought it should).

Reply:
Recursive is good :)